### ISSUE TITLE
Does CoreNeuron support the simulation of mod files with kinetic scheme or derivimplicit methods on GPU?

### ISSUE THREAD

**Issue by pkuzyc** [BlueBrain/CoreNeuron#269]
We have some mod files implemented with kinetic scheme, and failed to simulate the model with CoreNeuron on GPU. Does CoreNeuron support the simulation of these kinds of mechanisms? Is the python script kinderiv.py helpful? If so, could you please tell us how to use it? Thank you.

**Comment by pramodk** [comment]
@pkuzyc :  To generate optimised code for CPU/GPUs we have been working for new  MOD file to C++ converter called [NMODL](https://github.com/BlueBrain/nmodl). That supports all solvers including kinetic schemes. 

We haven't made new NMODL compiler as default but we already testing it with our models. We  haven't written detailed instructions with GPU backend yet. Do you have representative MOD file that we can test?

**Comment by pkuzyc** [comment]
Sorry, I mean NMOD file here. We use kinetic scheme in NMOD file (like the [HCN2](https://senselab.med.yale.edu/ModelDB/ShowModel?model=19176&file=%2fHCN2k%2fhcn2.mod) mechanism from ModelDB). Is it possible to simulate this kind of NMOD file with CoreNEURON on GPU?

**Comment by pkuzyc** [comment]
**Link errors when generating special-core** 
I can successfully generate "special-core" using nrnivmodl-core with some general NMOD files. However, when there are some NMOD files implemented with kinetic scheme, it failed. 

I used nrnivmodl-core with command:
`./nrnivmodl-core /path/to/nmodfiles`
and the error is:
```
coreneuron/coreneuron.cpp:
@E@nvlink error   : Undefined reference to '_ZN10coreneuron13sparse_threadEPNS_9SparseObjEiPiS2_PddiiiiS3_S2_PNS_11ThreadDatumEPNS_9NrnThreadEd' in 'x86_64/libcorenrnmech.a:ca_ch.o'
@E@nvlink error   : Undefined reference to '_ZN10coreneuron17_ss_sparse_threadEPNS_9SparseObjEiPiS2_PddiiiiS3_S2_PNS_11ThreadDatumEPNS_9NrnThreadEd' in 'x86_64/libcorenrnmech.a:hcn12_ch.o'
```

It seems that it's about the function sparse_thread? Could you please take a look at it? Thank you.

**Comment by cattabiani** [comment]
Just to mention that we are still working on this. The problem is not trivial, sorry

**Comment by pramodk** [comment]
I believe this issue has been fixed in #303.

(It's true that we haven't fixed the support in NMODL but that is not a blocker for @pkuzyc I believe).
